intrinsics.c

```diff
@@ -452,24 +464,37 @@ static int __gc_collect(size_t argc, struct cb_value *argv,
 static int pcall(size_t argc, struct cb_value *argv, struct cb_value *result)
 {
 	int failed;
-	ptrdiff_t sp;
 	struct cb_array *arr;
 
 	CB_EXPECT_TYPE(CB_VALUE_FUNCTION, argv[0]);
 	result->type = CB_VALUE_ARRAY;
 	arr = result->val.as_array = cb_array_new(2);
-	arr->len = 2;
 
-	sp = cb_vm_state.stack_top - cb_vm_state.stack;
 	failed = cb_value_call(argv[0], argv + 1, argc - 1, &arr->values[1]);
-	cb_vm_state.stack_top = cb_vm_state.stack + sp;
 
 	arr->values[0].type = CB_VALUE_BOOL;
 	arr->values[0].val.as_bool = !failed;
 	if (failed) {
 		assert(cb_error_p());
-		arr->values[1] = *cb_error_value();
+		arr->values[1] = cb_error_value();
```

Comment:
if it didn't fail let's set the second value to null rather than some undefined value (ignore if new arrays are initialized to null)

Reply:
if it didn't fail this is set to the value returned by the function